tests/integration/scrapers/scrapers-all-test.js

```diff
@@ -1,94 +1,172 @@
+  const dt = parts.filter(s => dateRegex.test(s));
+
+  const date = dt.length === 0 ? null : dt[0];
+  return {
+    scraperName: name,
+    date
+  };
+}
+
+/** Changes a URL to a filesystem-friendly name, removes extension. */
+function sanitizeUrl(s) {
+  const ext = path.extname(s);
+  return s
+    .replace(ext, '')
+    .replace(/^https?:\/\//i, '')
+    .replace(/[^a-z0-9]/gi, '_');
+}
+
+/** Run a single scraper test directory. */
+async function runTest(t, cacheRoot, testDirectory) {
+  const { scraperName, date } = scraperNameAndDateFromPath(testDirectory);
+
+  // Monkeypatch global get for this test.
+  // eslint-disable-next-line no-unused-vars
+  get.get = async (url, type, date, options) => {
+    const sanurl = sanitizeUrl(url);
+    const respFile = join(testDirectory, sanurl);
+    // console.log(`  Call: ${url}\n  Sanitized: ${sanurl}\n  Response: ${respFile}`);
+    return fs.readFile(join(cacheRoot, respFile));
+  };
+
+  const pathParts = [__dirname, '..', '..', '..', 'src', 'shared', 'scrapers', scraperName, 'index.js'];
+  const scraperObj = imports(join(...pathParts)).default;
+
+  let result = null;
+  try {
+    process.env.SCRAPE_DATE = date;
+    result = await runScraper.runScraper(scraperObj);
+  } catch (e) {
+    t.fail(`${scraperName} on ${date}, error scraping: ${e}`);
+  } finally {
+    delete process.env.SCRAPE_DATE;
+    get.get = oldGetGet;
   }
-  return d;
-};
 
-test('Scraper tests', async t => {
-  const testDirs = glob(join(shared, 'scrapers', '**', 'tests'), { onlyDirectories: true });
+  if (result) {
+    // Writing the actual scraper result so ppl can diff/investigate.
+    // These are ignored in .gitignore.
+    const actualFilepath = join(cacheRoot, testDirectory, 'actual.json');
+    await fs.writeJSON(actualFilepath, result, { log: false });
+
+    const expectedPath = join(cacheRoot, testDirectory, 'expected.json');
+    const fullExpected = await fs.readJSON(expectedPath);
+
+    // Ignore features (for now?).
+    const removeFeatures = d => {
+      delete d.feature;
+      return d;
+    };
+    const actual = JSON.stringify(result.map(removeFeatures));
+    const expected = JSON.stringify(fullExpected.map(removeFeatures));
+    const shortPath = cacheRoot.replace(process.cwd(), '');
+    const msg = `${scraperName} on ${date} (actual.json vs expected.json in ${shortPath}/${testDirectory})`;
+    t.equal(actual, expected, msg);
+  } else {
+    t.fail(`should have had a result for ${scraperName} on ${date}`);
+  }
+}
 
-  for (const testDir of testDirs) {
-    const scraperName = scraperNameFromPath(testDir); // e.g. `USA/AK`
-    process.env.OVERRIDE_CACHE_PATH = testDir;
+// Mutex
 
-    // dynamically import the scraper
-    // eslint-disable-next-line
-    const scraperObj = imports(join(testDir, '..', 'index.js'));
+// Each test folder modifies global state (get.get, and
+// process.env.SCRAPE_DATE), so we need to make sure that they're run
+// one at a time.  Global state is bad!
```

Comment:
yup. is there a plan to resolve this eventually?

Reply:
I don't think so, just yet.  It really should be passed to the scraper as an argument.